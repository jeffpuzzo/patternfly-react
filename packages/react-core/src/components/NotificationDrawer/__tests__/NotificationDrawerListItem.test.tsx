import React from 'react';
import { render, screen } from '@testing-library/react';
import { NotificationDrawerListItem } from '../NotificationDrawerListItem';

describe('NotificationDrawerListItem', () => {
  test('renders with PatternFly Core styles', () => {
    render(<NotificationDrawerListItem />);
    expect(screen.getByRole('listitem').outerHTML).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<NotificationDrawerListItem className="extra-class" />);
    expect(screen.getByRole('listitem').className).toMatchSnapshot();
  });

  test('drawer list item with isHoverable applied', () => {
    render(<NotificationDrawerListItem isHoverable />);
    expect(screen.getByRole('listitem').className).toContain('hoverable');
  });

  test('drawer list item with isRead applied', () => {
    render(<NotificationDrawerListItem isRead />);
    expect(screen.getByRole('listitem').className).toContain('read');
  });
});
