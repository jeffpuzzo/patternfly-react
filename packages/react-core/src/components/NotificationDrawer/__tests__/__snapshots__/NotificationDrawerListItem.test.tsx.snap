// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawerListItem className is added to the root element 1`] = `"pf-c-notification-drawer__list-item pf-m-hoverable pf-m-default extra-class"`;

exports[`NotificationDrawerListItem renders with PatternFly Core styles 1`] = `"<li class=\\"pf-c-notification-drawer__list-item pf-m-hoverable pf-m-default\\" tabindex=\\"0\\"></li>"`;
